planning_outlines/spectralcube-to-specutils/plan.md

```diff
@@ -0,0 +1,61 @@
+This describes a transition plan to move `spectral-cube` functionality to `specutils`.
+
+# Top-level plan
+
+1. Expand the "needs implementation" items into concrete issues in `specutils`.  In general new functionality that's cube-specific should be a function/class in a relevant space like `specutils.analysis.cube`.
+2. Write unit tests for all the exsting analysis/manipulation functions in `SpectrumCollection` and `Spectrum1D` with cube shape/wcs.
+3. Add a label "spectral-cube feature" for all those needed for feature parity.
+4. Create parallel Jira tickets to track and prioritize internal STScI work.
+5. Declare victory when the label is empty, and at that point suggest deprecation of `spectral-cube` to Adam Ginsburg.
+
+
+# Needs Implementation in specutils
+
+* `NDCube` compatibility (although a proof-of-concept is here: https://github.com/astrofrog/nddata-experiments) - note this has the *highest* uncertainty.
+* indexing via world coordinates.  This is already supported in `NDCube`, so re-using this functionality would be useful if possible.
+* `spectral_slab` - note however that this is another word for the existing spectral region machinery in `specutils.manipulation.extract_region`, just with slightly different syntax. So `spectral_slab` might just be an alternate name that would be adopted in a `RadioSpectrum1D` or similar.
+* Moment maps: Not this is already possible in `specutils`, particularly for first- (np.mean) and second- (line width) maps, but exposing a generic "nth moment" function is a good idea.
+* `minimal_subcube` (see spectral-cube docs), although this would need a different name in a `Spectrum1D` context (since it would also potentially mask off the spectral "edges"?)
+* Spectral extraction (based on astropy regions/photutils?)
+* `with_mask` convenience method (not mandatory, but spectral-cube demonstrates it's use as syntactic sugar?)
+* Lazy masking (may not be a requirement?)
+
+
+# Problems/incompatibilities and solutions
+
+## Problem: the name `Spectrum1D`/`SpectralCollection`
+
+A user may not naturally think "cube" when they see "1d" or even "collection".
+
+### Solution
+
+1. Documentation/explanation that yes, a cube can be "1D" in the sense that it's rectified.  Caveat: people don't read documentation!
+2. Set up some aliases or subclasses with specific conventions.  E.g. `specutils.RectifiedSpectralCube` could be a subclass of `Spectrum1D` but just with an initializer that checks that the wcs has a spatial and spectral component.
+3. Point users to `NDCube` as a way to see explicit cube language, but have them use the NDCube<->Spectrum1D compatibility layer as the way to still use `specutils` functionality
+
```

Comment:
Change or alias Spectrum1D to Spectrum or SpectrumND?

Reply:
That's a big change in `specutils`.  I'm not *against* this necessarily (I'm currently minorly against it but could be convinced), but just think it's better to have the discussion there.